PhoneAuth/Controllers/MealPlanViewController.swift

```diff
@@ -0,0 +1,267 @@
+            generateButton.widthAnchor.constraint(equalToConstant: 200),
+            generateButton.heightAnchor.constraint(equalToConstant: 50)
+        ])
+    }
+    
+    @objc private func closeTapped() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc private func generateMealPlan() {
+        activityIndicator.startAnimating()
+        let selectedMealType = mealTypes[mealTypeSegmentedControl.selectedSegmentIndex]
+        let selectedDiningHall = diningHalls[diningHallPicker.selectedRow(inComponent: 0)]
+        guard let generativeModel = generativeModel else {
+            print("GenerativeModel is not initialized")
+            return
+        }
+        // Convert selected meal type to API format
+        let mealCode: String
+        switch selectedMealType {
+        case "Breakfast":
+            mealCode = "breakfast"
+        case "Lunch":
+            mealCode = "lunch"
+        case "Dinner":
+            mealCode = "dinner"
+        default:
+            mealCode = "lunch"
+        }
+        
+        // Convert selected dining hall to API format
+        let diningCommonCode: String
+        switch selectedDiningHall {
+        case "Carrillo":
+            diningCommonCode = "carrillo"
+        case "De La Guerra":
+            diningCommonCode = "de-la-guerra"
+        case "Portola":
+            diningCommonCode = "portola"
+        default:
+            diningCommonCode = "ortega"
+        }
+        
+        // Get today's date in the required format
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        let today = dateFormatter.string(from: Date())
+        //let today = "2024-11-22"
+        // Show loading indicator
+        
+        
+        DiningAPIClient.shared.getMenu(date: today, diningCommonCode: diningCommonCode, mealCode: mealCode) { result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success(let entrees):
+                    let menuItems = entrees.map { $0.name }.joined(separator: "\n")
+                    
+                    //self.resultLabel.text = "Menu for \(selectedMealType) at \(selectedDiningHall):\n\n\(menuItems)"
+                    fetchFoodPrompt(for: menuItems, generativeModel: generativeModel) {
+                        [weak self] success, message in self?.showPopup(message: message)
+                            self?.activityIndicator.stopAnimating()
+                            if success {
+                                self?.showPopup(message: message)
+                            } else {
+                                self?.showPopup(message: "Error: \(message)")
+                            }
+                        
+                        
+                    }
+                    
+                case .failure(let error):
+                    self.resultLabel.text = "Error: \(error.localizedDescription)"
```

Comment:
Is this error thrown when dining commons are closed? It should display that as the error.

Reply:
It doesn't say it is closed, all it says is that the data couldn't be read because it wasn't in the correct format

Reply:
instead of displaying "Error: \(error.localizedDescription)" you can just display something like "Dining halls are closed" cause i think thats the only time the error occurs

Reply:
This error pops up when the dining common is closed I think. Can the popup show a msg for a user and not a developer? Ex: "Could not create meal since dining commons are closed"